### PR TITLE
Add/redux

### PR DESCRIPTION
**Pull request by kevinyen83**
Added Redux to the Adopt-a-Paw page and resolved errors in the test cases and related components.

### CODE REVIEW

File: app/GlobalRedux/Feautures/favorites/isFavoritesEmpty-slice.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { IsFavoritesEmptyState } from '../../../../types/favorites-types';
+import { PayloadAction } from '@reduxjs/toolkit';
+
+const initialState: IsFavoritesEmptyState = {
+  isFavoritesEmpty: true,
+};
+
```
Comment (Daniel54088) [inline app/GlobalRedux/Feautures/favorites/isFavoritesEmpty-slice.tsx:8]: I think you don't need create a whole new file for just another reducer state, just put them together.

initialState:{
favoritesItems: [],
isFavoritesEmpty: true,
lastId: 0,
.....
}

File: app/GlobalRedux/Feautures/pets/selectedPet-slice.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Pet, SelectedPetState } from '../../../../types/pet-types';
+import { PayloadAction } from '@reduxjs/toolkit';
+
+const initialState: SelectedPetState = {
+  selectedPet: null,
+};
+
```
Comment (Daniel54088) [inline app/GlobalRedux/Feautures/pets/selectedPet-slice.tsx:8]: ditto

File: app/GlobalRedux/Feautures/pets/pets-slice.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { Pet, PetsState } from '../../../../types/pet-types';
+import { PayloadAction } from '@reduxjs/toolkit';
+
+const initialState: PetsState = {
+  pets: [],
+};
```
Comment (Daniel54088) [inline app/GlobalRedux/Feautures/pets/pets-slice.tsx:7]: ditto

File: app/GlobalRedux/Feautures/pets/visiblePets-slice.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { VisiblePetsState } from '../../../../types/pet-types';
+import { PayloadAction } from '@reduxjs/toolkit';
+
+const initialState: VisiblePetsState = {
+  visiblePets: 12,
```
Comment (Daniel54088) [inline app/GlobalRedux/Feautures/pets/visiblePets-slice.tsx:6]: ditto

File: app/GlobalRedux/Feautures/popup/showFavorites-slice.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { FavoritesPopupState } from '../../../../types/popup-types';
+import { PayloadAction } from '@reduxjs/toolkit';
+
+const initialState: FavoritesPopupState = {
+  showFavorites: false,
```
Comment (Daniel54088) [inline app/GlobalRedux/Feautures/popup/showFavorites-slice.tsx:6]: ok so the same with rest of files. One file with one reducer states.

File: app/adopt_a_paw/page.tsx [hunk 6]
```diff
@@ -9,51 +9,57 @@ import '../../styles.css';
...
-  // Show more state
-  const [visiblePets, setVisiblePets] = useState<number>(12);
+  const pets = useAppSelector((state) => state.pets.pets);
+  const selectedPet = useAppSelector((state) => state.selectedPet.selectedPet);
+  const formSelectedPet = useAppSelector(
+    (state) => state.formSelectedPet.formSelectedPet
+  );
+  const category = useAppSelector((state) => state.category.category);
+  const isLoading = useAppSelector((state) => state.isLoading.isLoading);
+  const showFavorites = useAppSelector(
```
Comment (Daniel54088) [inline app/adopt_a_paw/page.tsx:37]: Redux has combineSelectors for combines them all to redce repeating code

File: constants/index.ts [hunk 7]
```diff
@@ -1,131 +1,131 @@
...
+    name: 'Mimi',
+    category: 'Cats',
     age: 15,
-    gender: "Female",
-    color: "White",
-    size: "Small",
-    location: "Glebe",
-    vaccination: "Yes",
-    availability: "Yes",
+    gender: 'Female',
```
Comment (Daniel54088) [inline constants/index.ts:28]: In real life or in the future when you work with others in the team, This small changes can separate to another PR to keep PR clean. Remember: one PR one thing.

Comment (kevinyen83) [reply constants/index.ts:28]: I see. Will follow your suggestion.

File: app/GlobalRedux/Feautures/favorites/isFavoritesEmpty-slice.tsx (hunk 1 above)
Comment (kevinyen83) [reply app/GlobalRedux/Feautures/favorites/isFavoritesEmpty-slice.tsx:8]: Done

File: app/GlobalRedux/Feautures/popup/showFavorites-slice.tsx (hunk 5 above)
Comment (kevinyen83) [reply app/GlobalRedux/Feautures/popup/showFavorites-slice.tsx:6]: Done

File: app/GlobalRedux/Feautures/pets/selectedPet-slice.tsx (hunk 2 above)
Comment (kevinyen83) [reply app/GlobalRedux/Feautures/pets/selectedPet-slice.tsx]: Done

File: app/GlobalRedux/Feautures/pets/visiblePets-slice.tsx (hunk 4 above)
Comment (kevinyen83) [reply app/GlobalRedux/Feautures/pets/visiblePets-slice.tsx]: Done

File: app/adopt_a_paw/page.tsx (hunk 6 above)
Comment (kevinyen83) [reply app/adopt_a_paw/page.tsx]: Done

File: package.json
Comment (kevinyen83) [inline package.json:1]: Sorry, @Daniel54088 , I forgot to remove the cypress installation from this commit... This installation is for another branch (add/cypress) and pull request, but won't affect the functionality.

Comment (Daniel54088) [reply package.json:1]: All good.

**Pull request merged**